### PR TITLE
internal: adjust application type to don't allow unknown fields

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: lib/command/types.d.ts [hunk 1]
```diff
@@ -230,11 +236,24 @@ type PrepareConfigsWithType<U extends ParseableConfigs> = Simplify<{
 /** Keep Options/Config filtered by choices */
 type OnlyChoices<D, C extends boolean> = D extends { choices: JHispterChoices } ? (C extends true ? D : never) : C extends true ? never : D;
 
-/** Keep Options/Config filtered by choices */
+/**
+ * Keep Options/Config filtered by choices
+ *
+ * @example
+ * ```ts
+ * type CofigsWithChoice = OnlyCofigsWithChoice<{ clientFramework: { choices: ['angular', 'no'], scope: 'storage' }, clientTestFramework: { choices: ['cypress', 'no'], scope: 'storage' } }>
```
Comment (DanielFran) [inline lib/command/types.d.ts:244]: Typo:
type ConfigsWithChoice = OnlyConfigsWithChoice<{

File: lib/command/types.d.ts [hunk 2]
```diff
@@ -230,11 +236,24 @@ type PrepareConfigsWithType<U extends ParseableConfigs> = Simplify<{
...
+ * ```
+ */
 type OnlyCofigsWithChoice<D extends ParseableConfigs, C extends boolean> = {
```
Comment (DanielFran) [inline lib/command/types.d.ts:247]: Typo 
type OnlyConfigsWithChoice<D extends ParseableConfigs, C extends boolean> = {

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: lib/command/types.d.ts (hunk 1 above)
Comment (mshima) [reply lib/command/types.d.ts:244]:
```suggestion
 * type ConfigsWithChoice = OnlyCofigsWithChoice<{ clientFramework: { choices: ['angular', 'no'], scope: 'storage' }, clientTestFramework: { choices: ['cypress', 'no'], scope: 'storage' } }>
```

File: lib/command/types.d.ts (hunk 2 above)
Comment (mshima) [reply lib/command/types.d.ts:247]:
```suggestion
type OnlyCofingsWithChoice<D extends ParseableConfigs, C extends boolean> = {
```

**Review by DanielFran — APPROVED** [review]

**Pull request merged**